crates/core/tedge_agent/src/agent.rs

```diff
@@ -46,13 +53,13 @@ use tracing::instrument;
 const TEDGE_AGENT: &str = "tedge-agent";
 
 #[derive(Debug, Clone)]
-pub struct AgentConfig {
+pub(crate) struct AgentConfig {
     pub mqtt_config: MqttConfig,
     pub http_config: HttpConfig,
+    pub http_socket_addr: SocketAddr,
```

Comment:
Why not keep this value also in the `HttpConfig` struct?

Reply:
In short, `HttpConfig` is used in a separate place to the `SocketAddr`, but I've figured out how I can get them to be used in a single place so that the tests aren't supplying information that isn't used.